### PR TITLE
add repo for 2.0.x

### PR DESCRIPTION
**Pull request by kubicgruenfeld**
#### Pull Request (PR) description
Adds support for the ProxySQL 2.0 repo.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -185,7 +185,8 @@
   Boolean $save_to_disk = $proxysql::params::save_to_disk,
 
   Boolean $manage_repo = true,
-  Hash $repo = $proxysql::params::repo,
+  Enum['14','20'] $repo_version = '14',
```
Comment (bastelfreak) [inline manifests/init.pp]: should this be an Enum on Integers and not strings?

File: types/galerahostgroup.pp [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# lint:ignore:2sp_soft_tabs
+type Proxysql::GaleraHostgroup = Array[Hash[String, Struct[{ writer                     => Integer,
```
Comment (bastelfreak) [inline types/galerahostgroup.pp]: datatypes \o/

**Comment by bastelfreak** [comment]
Hi @kubicgruenfeld, thanks for the PR! Can you have a look at the used email address in the commits? It isn't associated with your github account.

**Comment by kubicgruenfeld** [comment]
i have added the other mailaddress to my github account. also the pr needs some more work. i wanted at first only add the repo for 2.0.x but decided afterwards it is more reasonable to also add support for galera hostgroups and ubuntu 18.04 here. or should i do this in seperate PRs?

**Comment by bastelfreak** [comment]
seperate PRs would be cool. That makes it way easier for us to review it.

File: manifests/init.pp (hunk 1 above)
Comment (kubicgruenfeld) [reply manifests/init.pp]: i changed it. now it should definitely be a string.

File: types/galerahostgroup.pp (hunk 2 above)
Comment (kubicgruenfeld) [reply types/galerahostgroup.pp]: code removed. will be added in other PR

**Comment by kubicgruenfeld** [comment]
So i removed everything which is not related to the repo and will open other PRs for this.

File: manifests/repo.pp [hunk 3]
```diff
@@ -7,11 +7,35 @@
   if $proxysql::manage_repo == true {
     case $facts['os']['family'] {
       'Debian': {
-        create_resources('::apt::source', { 'proxysql_repo' => $proxysql::repo})
+        case $proxysql::repo_version {
+          '2.0.x': {
+            create_resources('::apt::source', { 'proxysql_repo' => $proxysql::repo20})
```
Comment (bastelfreak) [inline manifests/repo.pp]: please don't add the leading `::`. this is deprecated syntax and only required for puppet 3.

**Comment by bastelfreak** [comment]
thanks @kubicgruenfeld. Can you take a look at the inline comment and also update the README.md for the paramter and an explanation?

File: manifests/repo.pp (hunk 3 above)
Comment (kubicgruenfeld) [reply manifests/repo.pp]: i have removed the leading colons from all apt::source lines.

**Comment by kubicgruenfeld** [comment]
> thanks @kubicgruenfeld. Can you take a look at the inline comment and also update the README.md for the paramter and an explanation?

done. thanks for reviewing :)

**Comment by bastelfreak** [comment]
thanks! will merge it when travis gets green.

File: manifests/repo.pp [hunk 4]
```diff
@@ -7,11 +7,35 @@
...
-        create_resources('::apt::source', { 'proxysql_repo' => $proxysql::repo})
+        case $proxysql::repo_version {
+          '2.0.x': {
+            create_resources('apt::source', { 'proxysql_repo' => $proxysql::repo20})
+          }
+          '1.4.x': {
+            create_resources('apt::source', { 'proxysql_repo' => $proxysql::repo14})
+          }
+          default: {
+            create_resources('apt::source', { 'proxysql_repo' => $proxysql::repo14})
```
Comment (alexjfisher) [inline manifests/repo.pp]: Why `create_resources` when `Debian` and splat operator for `Redhat`?

**Pull request merged**